pkg/antctl/client.go

```diff
@@ -121,6 +128,9 @@ func (c *client) resourceRequest(e *resourceEndpoint, opt *requestOption) (io.Re
 	if err != nil {
 		return nil, err
 	}
+	if opt.server != "" {
+		kubeconfig.Host = opt.server
```

Comment:
Is this only useful for development (antrea components running as processes, no K8s apiserver)? I don't see another case where one would want to override kubeconfig with `--server`. Am I missing something?

Reply:
It's mainly for the case user override agent port due to port conflicts. The agent apiserver listens to a non default port so antctl will need a way to specify that port. The option is also available in kubectl, kept same style.

Reply:
I think my question wasn't clear. My understanding is that when we run antctl in "controller mode", antctl connects to the K8s apiserver, which thanks to the aggregation layer, forwards request to the Antrea Controller apiserver. kubeconfig tells us how to connect to the k8s apiserver, and the APIService definition in the yaml is used for the aggregation layer. So in "controller mode", it seems that one would have no use for `--server`. In "agent mode" on the other hand, `--server` is useful. My understanding is that - based on command registration - the`resourceRequest` method is only used in "controller mode". So is this code block useful here? And does `--server` indeed mean different things based on whether we are in "controller" or "agent" mode: k8s apiserver in "controller" mode, Antrea Agent apiserver in "agent" mode. If I understand correctly, we never connect *directly* to the Antrea Controller apiserver, since we use the aggregation layer for that.

Reply:
Just a side note not relevant to this change - you know we might need to disable API aggregation in some deployments like TKG, and in that case we might need antctl to support a different endpoint for Controller API as well, and we might also support auto-discover Controller ClusterIP/port from K8s API when antctl is executed inside the cluster.

Reply:
@antoninbas your understanding is completely correct, I added "-s" mainly for agent mode, but also saw kubelet has it anyway (perhaps its use cases applies to ours too) so thought it might be useful for development/troubleshooting purpose in controller mode too, otherwise we would have to update kubeconfig to debug antrea-controller during development. Also, the requestOption is a common struct to controller and agent mode so it's simpler to support both cases than making it an agent specific argument. 
Do you prefer not having it for controller?

Reply:
@jianjuns another side note you might already know: delegated authentication can only take token as credential so the default kubeconfig that uses client certs cannot be used to connect antrea-controller directly. User will need to generate a kubeconfig that has antctl serviceaccount token to connect. But perhaps this is right way in production.

Reply:
@tnqn I think you mean kubectl and not kubelet? Or maybe both have this `-s` flag but I think kubectl is a better comparison for antctl. I think keeping it as it is is fine, but I would recommend adding some more information (what we discussed here) to the comment for the `server` field in the `requestOption` struct? It's useful IMO to indicate which server we are actually connecting to in the different modes.

For my own understanding, could you clarify how this is useful for development / troubleshooting? If I follow these instructions (https://github.com/vmware-tanzu/antrea/blob/master/docs/manual-installation.md#antrea-controller), can I connect directly to the antrea-controller apiserver using the generated kubeconfig file and by providing the correct `-s` option (antrea-controller apiserver and not k8s apiserver)?

Reply:
basically, if we have to generate a kubeconfig with the correct ServiceAccount token anyway, we can include the correct `server` address directly (with the correct port number), and I am not sure what you mean by:

> otherwise we would have to update kubeconfig to debug antrea-controller during development

Reply:
Sure, will add comment to this field.
I meant typically one would generate a kubeconfig with antctl serviceaccount and antrea service ip for non aggregated apiserver case (if we have a guide or sample script, we probably will do so). So you could keep the config file as is and specify "-s" to override the server address when developing antrea-controller (running as a process), and be able to connect to antrea-controller directly when you suspect there's something wrong with service when troubleshooting, or when running multiple antrea-controllers in a cluster for comparison.

Reply:
updated comment, PTAL.